source/clear-linux/guides/maintenance/autospec.rst

```diff
@@ -1,189 +1,249 @@
+an existing spec file. The example uses the ``dmidecode`` package.
 
-You've successfully created a RPM.
+Make sure you have completed `Example 1: First time setup`_ before starting
+this example.
 
-Skip to `Next steps`_.
+#. Navigate to the autospec workspace and clone the ``dmidecode`` package.
 
-.. _build-source-code-with-existing-spec-file:
+   .. code-block:: bash
 
-Option 2: Build source code with an existing spec file
-======================================================
+      cd ~/clearlinux
+      make clone_dmidecode
 
-Use this method if you only want to build the RPM using the spec file. This
-method assumes that a spec file already exists. In this example, we run a
-``make build`` on the ``dmidecode`` package.
+   Or you can clone all package repos at once using:
 
-#. Navigate to the ``dmidecode`` package in clearlinux:
+   .. code-block:: bash
+
+      make [-j NUM] clone-packages
+
+#. Navigate to the local copy of the ``dmidecode`` package and build it:
 
    .. code-block:: bash
 
       cd ~/clearlinux/packages/dmidecode/
+      make build
+
+#. The resulting RPMs are in :file:`./rpms`. Logs are in :file:`./results`.
 
-#. To download the tarball and build, run the command:
+Example 3: Build a new RPM
+==========================
+
+This example shows how to build a new RPM with no spec file. The example will
+create a simple helloclear RPM.
+
+Make sure you have completed `Example 1: First time setup`_ before starting
+this example.
+
+#. Navigate to the autospec workspace and build the helloclear RPM. The
+   :file:`Makefile` provides a :command:`make autospecnew` that can
+   automatically generate an RPM package using the autospec tool. You must pass
+   the URL to the source tarball and the NAME of the RPM you wish to create:
 
    .. code-block:: bash
 
-      make build
+      cd ~/clearlinux
+      make autospecnew URL="https://github.com/clearlinux/helloclear/archive/helloclear-v1.0.tar.gz" NAME="helloclear"
 
-**Congratulations!**
+#. The resulting RPMs are in :file:`./rpms`. Logs are in :file:`./results`.
 
-You've successfully created a RPM.
+#. If build failures or dependency issues occur, provide the necessary
+   dependency, ban, or exclusion information via control files to autospec:
 
-Skip to `Next steps`_.
+   #. Navigate to the specific package.
 
-.. _generate-a-new-spec-file:
+      .. code-block:: bash
 
-Option 3: Generate a new spec file with a pre-defined package
-=============================================================
+         cd ~/clearlinux/packages/<package-name>
 
-Use this method to modify an existing package. In this example, you will
-modify an existing |CL| package called ``dmidecode`` to create a custom
-RPM. You will make a simple change to this package, change the revision to
-a new number that is higher than the |CL| OS version, and rebuild the package.
+   #. Respond to the build process output by editing control files to resolve
```

Comment:
This step doesn't seem to make sense as part of this example because "helloclear" builds without issue.  It's more fitting to use it as general workflow or create another example that requires touching the control files like the "opae-sdk" example.